docs/DeveloperGuide.md

```diff
@@ -296,6 +296,151 @@ The following sequence diagram shows how the flow of the example execution:
+
+Each `ListEntry` object contains a list of `ListEntryField` objects, which are the fields of the `ListEntry` object. For example, a `Person` object contains a list of `ListEntryField` objects, which are the fields of the `Person` object, such as name, phone number, email address, etc.
+
+These two classes are used in multiple generic class and methods to achieve the flexibility and robustness of the parsing process.
+
+##### 2. Parsing of Flags and Parameters
+The `TypeParsingUtil` class parse all kinds of flags (which are all `ListEntryField` ) via a single powerful generic method, `parseField`. The method signature is as follows:
+```
+ public static <T extends ListEntryField> T parseField(String flagName,
+                                                          String input,
+                                                          Of<T> of,
+                                                          boolean isOptional) throws ParseException
+```
+The first parameter, `flagName`, is the name of the flag that is being parsed. It will also be used to generate the error message when the parsing fails.
+
+The second parameter, `input`, is the user input that is intended to be parsed into the flag.
+
+The third parameter, `of`, is a self-defined functional interface that is to be used as the "factory" to create the flag object. The `of` method signature is as follows:
+```
+public interface Of<T extends ListEntryField> {
+    T apply(String str) throws IllegalArgumentException, ParseException;
+}
+```
+The last parameter, `isOptional`, is a boolean flag that indicates whether the flag is optional or not. If the flag is not found in the user input, this method will throw a `ParseException` if the flag is not optional, but will return a `null` object if the flag is optional. It is overloaded to has a default value of `true` when omitted.
+
+
+The `parseField` method is reused extensively in multiple parser classes to parse all the flags. The following is an example of how it can be used:
+```
+// Parse the name flag, which is compulsory
+Name name = TypeParsingUtil.parseField("name", args, Name::of, false);
+
+// Parse the phone flag, which is optional
+Phone phone = TypeParsingUtil.parseField("phone", args, Phone::of);
+```
+##### 3. Parsing of the EditPersonCommand
+The `EditPersonCommandParser` class will parse the user input into an `EditPersonCommand` object. 
+
+This is achieved by combining the results of parsing flags in the previous stage and construct a `Person` object that contains the updated fields.
+
+Each of the `ListEntryField` has a global and static singleton default instance. For example, the default `Phone` is a `Phone` object with message "To be added."Please also note that it is not possible to create a `Phone` object with such message via public constructor and factory method.
```

Comment:
I think can add a space between "To be added." and Please

Reply:
Thank you for pointing that out. Fixed in the latest commit.